Source/Lib/Codec/EbEncHandle.c

```diff
@@ -3621,42 +3710,72 @@ static EB_ERRORTYPE CopyFrameBuffer(
+            unpackDataPtr->out8BitBuffer   = inputPicturePtr->bufferY + lumaBufferOffset + (EB_U32)(out_luma/nb_luma)*i;
+            unpackDataPtr->out8Stride      = inputPicturePtr->strideY;
+            unpackDataPtr->outnBitBuffer   = inputPicturePtr->bufferBitIncY + lumaBufferOffset + (EB_U32)(out_luma/nb_luma)*i;
+            unpackDataPtr->outnStride      = inputPicturePtr->strideBitIncY;
+            unpackDataPtr->width           = lumaWidth;
+            unpackDataPtr->height          = lumaHeight/nb_luma;
+            EbPostFullObject(copyFrameBufferWrapperPtr);
+        }
+
+        for(int i = 0;i< nb_chroma; i++){
+            EbGetEmptyObject(context_unpack->copyFrameInputFifoPtr,&copyFrameBufferWrapperPtr);
+            unpackDataPtr                      = (EB_ENC_UnPack2D_TYPE_t*)copyFrameBufferWrapperPtr->objectPtr;
+            unpackDataPtr->in16BitBuffer       = (EB_U16*)(inputPtr->cb) + (EB_U32)(in_chroma/nb_chroma)*i;
+            unpackDataPtr->inStride            = (EB_U16)(inputPtr->cbStride);
+            unpackDataPtr->out8BitBuffer       = inputPicturePtr->bufferCb + chromaBufferOffset + (EB_U32)(out_chroma/nb_chroma)*i;
+            unpackDataPtr->out8Stride          = inputPicturePtr->strideCb;
+            unpackDataPtr->outnBitBuffer       = inputPicturePtr->bufferBitIncCb + chromaBufferOffset + (EB_U32)(out_chroma/nb_chroma)*i;
+            unpackDataPtr->outnStride          = inputPicturePtr->strideBitIncCb;
+            unpackDataPtr->width               = chromaWidth;
+            unpackDataPtr->height              = chromaHeight/nb_chroma;
+            EbPostFullObject(copyFrameBufferWrapperPtr);
+        }
+
+        for(int i = 0;i< nb_chroma;i++){
+            EbGetEmptyObject(context_unpack->copyFrameInputFifoPtr,&copyFrameBufferWrapperPtr);
+            unpackDataPtr                      = (EB_ENC_UnPack2D_TYPE_t*)copyFrameBufferWrapperPtr->objectPtr;
+            unpackDataPtr->in16BitBuffer       = (EB_U16*)(inputPtr->cr) + (EB_U32)(in_chroma/nb_chroma)*i;
+            unpackDataPtr->inStride            = (EB_U16)(inputPtr->crStride);
+            unpackDataPtr->out8BitBuffer       = inputPicturePtr->bufferCr + chromaBufferOffset  + (EB_U32)(out_chroma/nb_chroma)*i;
+            unpackDataPtr->out8Stride          = inputPicturePtr->strideCr;
+            unpackDataPtr->outnBitBuffer       = inputPicturePtr->bufferBitIncCr + chromaBufferOffset  + (EB_U32)(out_chroma/nb_chroma)*i;
+            unpackDataPtr->outnStride          = inputPicturePtr->strideBitIncCr;
+            unpackDataPtr->width               = chromaWidth;
+            unpackDataPtr->height              = chromaHeight/nb_chroma;
+            EbPostFullObject(copyFrameBufferWrapperPtr);
+        }
+        
+        EbObjectWrapper_t   *unpackEndSyncWrapperPtr;
+        for(int i = 0 ; i < nb; i++){
+            EbGetFullObject(context_unpack->unPackOutPutFifoPtr,&unpackEndSyncWrapperPtr);
```

Comment:
Can we wait for the object posted from the unpackTasksConsumerFifoPtrArray queue, to make sure the buffers are completely unpacked (synced)? I mean, maybe we don't need the redundant unpackSync SystemResource and producer/consumer queues? Thanks!

Reply:
I think we can't wait for it in CopyFrameBuffer because the unpackTasksConsumerFifo is a one way queue to deliver the datas to unpack function. We need a synchronization between CopyFrameBuffer and the threads so the unpackSyncConsumerFifo is doing the sync.